lucene/core/src/java/org/apache/lucene/search/WANDScorer.java

```diff
@@ -319,7 +356,10 @@ private void pushBackLeads(int target) throws IOException {
       }
     }
     lead = null;
-    leadMaxScore = 0;
+
+    if (needsScore) {
+      leadMaxScore = 0;
+    }
```

Comment:
I wonder if this condition could be removed in order to make this class easier to understand by minimizing differences in code paths between the scoring and the non-scoring cases? Are there other branches we could remove?

Reply:
Makes sense. Let me take a further look at this and try out.

Reply:
In the latest commit, I've done some refactoring and removed a few newly added branches that can be no-op for non-scoring mode, but not by a lot. Currently there are about 5 conditional branches left for handling non-scoring cases (checking `scoreMode  == ScoreMode.TOP_SCORES`), 1 is used in constructor, 2 used in assertions, and the last 2 were used in `advance` and `doNextCompetitiveCandidate`, so I felt that might be the minimal number of branches we could have now without major refactoring changes? What do you think?